scripts/benchmark.py

```diff
@@ -0,0 +1,341 @@
+"""Weave Benchmark Runner
+
+A typer app for selecting and running benchmarks from the benchmarks/ directory.
+"""
+#!/usr/bin/env -S uv run --script
+# /// script
+# dependencies = [
+#     "rich==14.0.0",
+#     "typer==0.16.0",
+# ]
+# ///
```

Comment:
Is this necessary?

Reply:
Yes, this is what makes it a self-contained uv script.
See: https://peps.python.org/pep-0723/